admin/services/data-access/pupils-not-taking-check.data.service.js

```diff
@@ -2,27 +2,55 @@
 
 const Pupil = require('../../models/pupil')
 const AttendanceCode = require('../../models/attendance-code')
-const pupilsNotTakingCheckDataService = {}
+const sqlService = require('./sql.service')
+const TYPES = require('tedious').TYPES
 
+const pupilsNotTakingCheckDataService = {
 /**
- *
  * @param schoolId
+ * @deprecated use sqlFindPupilsWithReasons
  * @returns {Promise.<*>}
  */
-pupilsNotTakingCheckDataService.fetchPupilsWithReasons = async (schoolId) => {
-  return Pupil
-    .find({'attendanceCode': {$exists: true}, 'school': schoolId})
-    .sort('lastName')
-    .lean()
-    .exec()
-}
+  fetchPupilsWithReasons: async (schoolId) => {
+    return Pupil
+      .find({'attendanceCode': {$exists: true}, 'school': schoolId})
+      .sort('lastName')
+      .lean()
+      .exec()
+  },
+  /**
+ * @param schoolId
+ * @description returns all pupils with specified school that have a record of attendance
+ * @returns {Promise.<*>}
+ */
+  sqlFindPupilsWithReasons: async (schoolId) => {
+    const sql = `SELECT DISTINCT p.* FROM ${sqlService.adminSchema}.[pupil] p 
+    INNER JOIN ${sqlService.adminSchema}.[pupilAttendance] pa
+    ON p.id = pa.pupil_id WHERE p.school_id = @schoolId`
+    const params = [{
+      name: 'schoolId',
+      value: schoolId,
+      type: TYPES.Int
+    }]
+    return sqlService.query(sql, params)
+  },
+  /**
+ *
+ * @deprecated use sqlFindAttendanceCodes
+ * @returns {Promise.<*>}
+ */
+  getAttendanceCodes: async () => {
+    return AttendanceCode
+      .find()
+      .sort('order')
+      .lean()
+      .exec()
+  },
+  sqlFindAttendanceCodes: async () => {
```

Comment:
This seems to be in the wrong service.  Shouldn't it be in the `attendanceCodeDataService`?  Leaving it in the `pupils-not-taking-check.data.service` means that dependencies that are not needed would be pulled in by the calling module.

Reply:
AttendanceCodeDataService currently doesn't exist right?
Extracting a new service which indeed does make sense would mean doing more than what the ticket is about.
It does seem a small change however so why not.

Reply:
@jon-shipley @GeorgeChatzigiannis yes its an odd set of functions in the one data service.  plus, the other attendance status functions are scattered around, some of which i believe are in other pupil related data services. messy

Reply:
`attendanceCodeDataService` created